code/modules/movespeed/modifiers/reagent.dm

```diff
@@ -0,0 +1,42 @@
+/datum/movespeed_modifier/reagent
+	blacklisted_movetypes = (FLYING|FLOATING)
+
+/datum/movespeed_modifier/reagent/stimulants
+	id = "stimulants_reagent"
```

Comment:
Can't you just use the type for most things and only have a text id when multiple types need to share one id?